app/lib/actions/helpers/lock.rb

```diff
@@ -4,19 +4,24 @@ module Lock
       # @see Lock.exclusive!
       def exclusive_lock!(resource)
         phase! Dynflow::Action::Plan
-        ::ForemanTasks::Lock.exclusive!(resource, task.id)
+        parent_lock = ::ForemanTasks::Lock.for_resource(resource).where(:task_id => task.self_and_parents.map(&:id)).first
+        if parent_lock
+          ForemanTasks::Link.link_resource_and_related!(resource, task)
+          parent_lock
+        else
+          ::ForemanTasks::Lock.exclusive!(resource, task)
+        end
       end
 
       # @see Lock.lock!
-      def lock!(resource, *lock_names)
-        phase! Dynflow::Action::Plan
-        ::ForemanTasks::Lock.lock!(resource, task.id, *lock_names.flatten)
+      def lock!(resource, *_lock_names)
```

Comment:
Do I read it correctly we are renaming this to `exclusive_lock!` to be more verbose about the behaviour?
Should we deprecate this method?

Reply:
The rename was done because of the change in arity. Guess deprecation warning would be a good idea

Reply:
Marked as deprecated and to be removed in foreman 2.3, hope this gives enough time for people to adjust to this change.

Reply:
2.3 is around a corner :thinking: shouldn't we enable one version in production?

Reply:
Well, my thought was that 2.2 is not even out yet, but I wouldn't mind bumping it to 2.4

Reply:
I'd be for bumping to 2.4, because this goes to production with foreman 2.3, right? we won't be backporting it to 2.2, even though it still in RC, right? :shrug:

Reply:
Right, 2.4 it is